File: python_practice/11.盛最多水的容器.py
#
# @lc app=leetcode.cn id=11 lang=python3
#
# [11] 盛最多水的容器
#
from typing import List

# @lc code=start
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right = 0, len(height) - 1
        record = 0
        while left < right:
            area = (right - left) * min(height[left], height[right])  
            if area > record:
                record = area
            if height[left] > height[right]:
                right -= 1
            else:
                left += 1
        
        return record
# @lc code=end

File: cpp/11.盛最多水的容器.cpp
/*
 * @lc app=leetcode.cn id=11 lang=cpp
 *
 * [11] 盛最多水的容器
 */
#include <vector>
using namespace std;

// @lc code=start
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size() - 1;
        int record = 0;
        int area = 0;
        while (left < right)
        {
            area = std::min(height[right], height[left]) * (right - left);
            if (area > record)
                record = area;

            if (height[left] > height[right]) {
                right--;
            } else {
                left++;
            }
        }

        return record;
    }
};
// @lc code=end

